### ISSUE TITLE
Add CreateMany

### ISSUE THREAD

**Issue by steebchen** [prisma/prisma-client-go#548]

**Comment by MaximilianGaedig** [comment]
Hi, is this a good first issue? I would love to give it a try, do you have some directions on where to get started?

**Comment by steebchen** [comment]
Kind of, yeah. All generator related code is not the easiest to grasp though, and there are no docs for Prisma AST. There are some tricks to see the info etc. though and I could write up some doc, but it definitely takes some time to understand the concepts and how the Prisma internals work (which are not optimized for Go but rather JS).


I'm also not sure about the priority of this, as you can create many docs using transactions right now:

```go
createUserA := client.User.CreateOne(
	User.Email.Set("a"),
	User.ID.Set("a"),
).Tx()

createUserB := client.User.CreateOne(
	User.Email.Set("b"),
	User.ID.Set("b"),
).Tx()

if err := client.Prisma.Transaction(createUserA, createUserB).Exec(ctx); err != nil {
	t.Fatal(err)
}
```

This would also work with a loop or X docs:

```go
var txs []transaction.Param

for item := someItems {
  tx := client.User.CreateOne(
	User.Email.Set("b"),
	User.ID.Set("b"),
  ).Tx()
  txs = append(txs, tx)
}

if err := client.Prisma.Transaction(txs...).Exec(ctx); err != nil {
	t.Fatal(err)
}
```

**Comment by MaximilianGaedig** [comment]
Hmm yeah.. that's what I am doing right now, problem is the memory usage on that, a transaction is probably quite big, and when passing it to the prisma intermediate service memory usage is horrendous:
![image](https://github.com/steebchen/prisma-client-go/assets/38767445/6fd6337a-bd84-4f24-a83d-bfc8fc214179)
I am hoping this could reduce the memory usage. Or is it just a prisma service issue handling the usecase of inserting ~70k records a lot of times without going above crazy memory usage

Even in my process all the transactions when batching them in batches of 20k I experience 200MB of memory usage on that alone (which would be okay, but far from what CreateMany could achieve with instead of adding transactions to a list we could add Models to the list for example)

**Comment by MaximilianGaedig** [comment]
Syntax wise I imagined it to be something like this:
```go
userA := db.User{ id:"a", email:"a" }
userB := db.User{ id:"b", email:"b" }

if err := client.User.CreateMany(userA, userB).Exec(ctx); err != nil {
  log.Fatal(err)
}
```

**Comment by steebchen** [comment]
Well, internally a CreateMany would also run in a transaction, but potentially it's more optimized, but I'm not sure. Since you are saying "what CreateMany could achieve" – did you actually test this? You could do so with the JS client and see if CreateMany is faster than CreateOne with transactions.

Regarding the syntax, this would not really work, as the syntax would be different from CreateOne and it would not benefit from the extra type-safety (think about also linking records).

**Comment by MaximilianGaedig** [comment]
Hmm okay, will try to check it out in the JS client, this was just an assumption from me for now

**Comment by sx328** [comment]
@MaximilianGaedig Did you get a chance to see if it was faster for transactions?

**Comment by MaximilianGaedig** [comment]
Not yet, but it's on my TODO list, I needed more control over the database for the project I used this in anyways tho so I went to pgx with that

**Comment by sx328** [comment]
I see, no worries

**Comment by steebchen** [comment]
@nettrino Yes this would work, with the disadvantage that it doesn't run in a transaction. So it might work depending on the use-case.

**Comment by nettrino** [comment]
@steebchen  my understanding is that a transaction is an all-or-nothing operation, where statements are executed in order as passed, whereas my above suggestion is just for records that could be created asynchronously as needed without depending on one another - is that assumption correct?
